src/test/java/org/owasp/html/AutoCloseableHtmlStreamRendererTest.java

```diff
@@ -1,85 +0,0 @@
-package org.owasp.html;
-
-import java.io.Closeable;
-import java.io.IOException;
-import java.lang.reflect.InvocationHandler;
-import java.lang.reflect.Method;
-import java.lang.reflect.Proxy;
-import java.util.IdentityHashMap;
-import java.util.Map;
-
-import org.junit.Test;
-
-import junit.framework.TestCase;
-
-@SuppressWarnings("javadoc")
-public class AutoCloseableHtmlStreamRendererTest extends TestCase {
-
-  static final class SimpleCloseable implements Closeable {
-    boolean closed;
-
-    public void close() throws IOException {
-      this.closed = true;
-    }
-  }
-
-
-  @Test
-  public static void testThatCloseablesAreClosed() throws IOException {
-    @SuppressWarnings("resource")
-    SimpleCloseable closeable = new SimpleCloseable();
-
-    assertFalse(closeable.closed);
-
-    assertTrue(AutoCloseableHtmlStreamRenderer.isAutoCloseable(closeable));
-
-    assertFalse(closeable.closed);
-
-    AutoCloseableHtmlStreamRenderer.closeIfAnyCloseable(closeable);
-
-    assertTrue(closeable.closed);
-  }
-
-  @Test
-  public static void testThatAutoCloseablesAreClosed() throws IOException {
```

Comment:
Do we have any other tests that closing an autocloseable closes the underlying appendable?

Reply:
THe original code (to which I reverted), also had to tests.

Reply:
How is this expected to work?

Take the following code:
```Java
package org.owasp.html;

import java.io.IOException;

import junit.framework.TestCase;

@SuppressWarnings("javadoc")
public class AutoCloseableHtmlStreamRendererTest extends TestCase {
	  public static void testBlup() {
	    PolicyFactory policyBuilder = new HtmlPolicyBuilder()
	      .allowAttributes("href").onElements("a")
	      .allowStandardUrlProtocols()
	      .allowElements("a")
	      .toFactory();

	    AutoClosableAppender autoClosableAppender = new AutoClosableAppender();
	    assertFalse(autoClosableAppender.closed);
		HtmlStreamRenderer renderer = HtmlStreamRenderer.create(autoClosableAppender,
				Handler.DO_NOTHING,
				Handler.DO_NOTHING);
		// Use the policy defined above to sanitize the HTML.
		HtmlSanitizer.sanitize("<a href=http://example.com>A link</a>", policyBuilder.apply(renderer));
		assertTrue(autoClosableAppender.closed);
	  }

	private static class AutoClosableAppender implements Appendable, AutoCloseable {
		public boolean closed = false;
		@Override
		public void close() throws Exception {
			closed = true;
		}

		@Override
		public Appendable append(CharSequence csq) throws IOException {
			return this;
		}

		@Override
		public Appendable append(CharSequence csq, int start, int end) throws IOException {
			return this;
		}

		@Override
		public Appendable append(char c) throws IOException {
			return this;
		}
	}
}
```

Close is never called (tested with: e8aa0f1b75ab81eea97ef173fefc023a3e16963d (current main) and eda1118b8c48f9e04ee916bec49102851bcfe943)
Also HtmlStreamRenderer.CloseableHtmlStreamRenderer.close seems to be never called

Reply:
Yeah.  Java has intersection types but doesn't allow mentioning them.

Reply:
If this is the intended usage, maybe we need another factory method and make the inner-classes public